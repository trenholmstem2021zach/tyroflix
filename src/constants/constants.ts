import { Feature } from "../components";

export const IMAGE_BASE_URL = 'https://image.tmdb.org/t/p/';
export const IMAGE_SIZES = {
	backdrop: {
		small: 'w300',
		xmedium: 'w780',
		medium: 'w1280',
		large: 'original'
	},
	poster: {
		small: 'w154',
		medium: 'w185',
		large: 'w342'
	},
	still: {
		small: 'w92'
	}
};
export const BACKDROP_PLACEHOLDER = 'https://plchldr.co/i/300x169?bg=333333&text=TyroFlix';
export const POSTER_PLACEHOLDER = 'https://plchldr.co/i/342x513?bg=333333&text=TyroFlix';
export const STILL_PLACEHOLDER = 'https://plchldr.co/i/185x104?bg=333333&text=TyroFlix';

export const HEADER_LINKS = [
	{
		title: 'TV Shows',
		value: 'series'
	},
	{
		title: 'Movies',
		value: 'movies'
	}
];

export const FAQS = [
	{
		id: 1,
		header: 'What is TyroFlix? Jamar',
		body:
			"TyroFlix is a FREE streaming service that offers tutorial and learning media to help you grow your skills to get the career you want.  There's always something new to discover, and new TV shows and movies are added every week!"
	},
	{
		id: 2,
		header: 'How much does TyroFlix cost?',
		body:
			'Watch TyroFlix on your smartphone, tablet, Smart TV, laptop or streaming device, all for FREE, FREE , yes FREE'
	},
	{
		id: 3,
		header: 'Where can I watch? Charlie',
		body:
			"Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Tyroflix account to watch instantly on the web at tyroflix.com from your personal computer or on any internet-connected device that offers the TyroFlix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.\n\nYou can also download your favourite shows with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take TyroFlix with you anywhere."
	},
	{
		id: 4,
		header: 'How do I cancel?  Zach',
		body:
			'TyroFlix is flexible. it is free , and we provide an easy delete account Feature.'
	},
	{
		id: 5,
		header: 'What can I watch on TyroFlix?',
		body:
			'TyroFlix has an extensive library of tutorials, documentaries, TV shows, award-winning TyroFlix originals, and more. Watch as much as you want, any time you want.'
	}
];

export const FEATURES = [
	{
		id: 1,
		title: 'Enjoy on your TV.',
		description: 'Watch on smart TVs, PlayStation, Xbox, Chromecast, Apple TV, Blu-ray players and more.',
		video: '',
		image: '/images/misc/Home-TV.jpg',
		alt: 'TyroFlix on TV'
	},
	{
		id: 2,
		title: 'Download your shows to watch offline.',
		description: 'Save your favorites easily and always have something to watch.',
		video: '',
		image: '/images/misc/Home-Mobile.jpg',
		alt: 'TyroFlix on mobile',
		direction: 'row-reverse'
	},
	{
		id: 3,
		title: 'Watch everywhere.',
		description: 'Stream unlimited movies and TV shows on your phone, tablet, laptop and TV without paying more.',
		image: '/images/misc/Home-IMac.jpg',
		alt: 'TyroFlix on different devices'
	}
];

export const FOOTER_LINKS = [
	{ id: 1, text: 'FAQ', url: 'https://help.TyroFlix.com/support/412' },
	{ id: 2, text: 'Help Center', url: 'https://help.TyroFlix.com/' },
	{ id: 3, text: 'Account', url: 'https://www.TyroFlix.com/youraccount' },
	{ id: 4, text: 'Media Center', url: 'https://media.TyroFlix.com/' },
	{ id: 5, text: 'Investor Relations', url: 'http://ir.TyroFlix.com/' },
	{ id: 6, text: 'Jobs', url: 'https://jobs.TyroFlix.com/jobs' },
	{ id: 7, text: 'Redeem Gift Cards', url: 'https://www.TyroFlix.com/redeem' },
	{ id: 8, text: 'Buy Gift Cards', url: 'https://www.TyroFlix.com/gift-cards' },
	{ id: 9, text: 'Ways to Watch', url: 'https://www.TyroFlix.com/watch' },
	{ id: 10, text: 'Terms of Use', url: 'https://help.TyroFlix.com/legal/termsofuse' },
	{ id: 11, text: 'Privacy', url: 'https://help.TyroFlix.com/legal/privacy' },
	{ id: 12, text: 'Cookie Preferences', url: 'https://www.TyroFlix.com/ph/#' },
	{ id: 13, text: 'Corporate Information', url: 'https://help.TyroFlix.com/legal/corpinfo' },
	{ id: 14, text: 'Contact Us', url: 'https://help.TyroFlix.com/contactus' },
	{ id: 15, text: 'Speed Test', url: 'https://fast.com/' },
	{ id: 16, text: 'Legal Notices', url: 'https://help.TyroFlix.com/legal/notices' },
	{ id: 17, text: 'TyroFlix Originals', url: 'https://www.TyroFlix.com/ph/browse/genre/839338' }
];
