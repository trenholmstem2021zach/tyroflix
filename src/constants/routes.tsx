import { Home, Browse, Signin, Signup } from '../pages';
import PPolicy from '../pages/PPolicy';

type Routes = {
	[key: string]: {
		path: string;
		page: JSX.Element;
		isProtected: boolean;
	};
};

export const ROUTES: Routes = {
	HOME: { path: '/', page: <Home />, isProtected: false },
	PRIVACYPOLICY: { path: '/privacy', page: <PPolicy />, isProtected: false },
	BROWSE: { path: '/browse', page: <Browse />, isProtected: true },
	SIGNUP: { path: '/signup', page: <Signup />, isProtected: false },
	SIGNIN: { path: '/signin', page: <Signin />, isProtected: false }
};
