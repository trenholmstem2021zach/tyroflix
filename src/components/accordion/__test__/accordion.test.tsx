import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Accordion from '../index';
import { FAQS } from '../../../constants/constants';

describe('<Accordion />', () => {
	it('renders the <Accordion /> with populated data', () => {
		const { container, getByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{FAQS.map(({ id, header, body }) => (
					<Accordion.Item key={id} id={id}>
						<Accordion.Header id={id}>{header}</Accordion.Header>
						<Accordion.Body id={id}>{body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);

		expect(getByText('Frequently Asked Questions')).toBeTruthy();
		expect(getByText('What is Roseflix?')).toBeTruthy();
		expect(getByText('How much does Roseflix cost?')).toBeTruthy();
		expect(getByText('Where can I watch?')).toBeTruthy();
		expect(getByText('How do I cancel?')).toBeTruthy();
		expect(getByText('What can I watch on Roseflix?')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('opens and closes the <Accordion /> component', () => {
		const { container, queryByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{FAQS.map(({ id, header, body }) => (
					<Accordion.Item key={id} id={id}>
						<Accordion.Header id={id}>{header}</Accordion.Header>
						<Accordion.Body id={id}>{body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);

		// TODO expect body to be toggled
		// const whatIsRoseflixBodyText =
		// 	"Roseflix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries and more on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single commercial â€“ all for one low monthly price. There's always something new to discover, and new TV shows and movies are added every week!";
		// const whatIsRoseflixHeaderEl = queryByText('What is Roseflix?');
		// expect(queryByText(whatIsRoseflixBodyText)).toBeFalsy();
		// if (whatIsRoseflixHeaderEl) fireEvent.click(whatIsRoseflixHeaderEl);
		// expect(queryByText(whatIsRoseflixBodyText)).toBeTruthy();
		// if (whatIsRoseflixHeaderEl) fireEvent.click(whatIsRoseflixHeaderEl);
		// expect(queryByText(whatIsRoseflixBodyText)).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
