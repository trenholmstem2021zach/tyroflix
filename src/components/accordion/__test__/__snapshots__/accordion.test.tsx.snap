// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-11jpbmb-0 iKHJax"
>
  <div
    class="accordion__Inner-sc-11jpbmb-2 kjxBzO"
  >
    <h1
      class="accordion__Title-sc-11jpbmb-1 kziJvb"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        What is Roseflix?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries and more on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single commercial – all for one low monthly price. There's always something new to discover, and new TV shows and movies are added every week!
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        How much does Roseflix cost?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Watch Roseflix on your smartphone, tablet, Smart TV, laptop or streaming device, all for one fixed monthly fee. Plans range from ₱149 to ₱549 a month. No extra costs, no contracts.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        Where can I watch?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Roseflix account to watch instantly on the web at roseflix.com from your personal computer or on any internet-connected device that offers the Roseflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite shows with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Roseflix with you anywhere.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        How do I cancel?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix is flexible. There are no pesky contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account anytime.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        What can I watch on Roseflix?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix has an extensive library of feature films, documentaries, TV shows, anime, award-winning Roseflix originals, and more. Watch as much as you want, any time you want.
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-11jpbmb-0 iKHJax"
>
  <div
    class="accordion__Inner-sc-11jpbmb-2 kjxBzO"
  >
    <h1
      class="accordion__Title-sc-11jpbmb-1 kziJvb"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        What is Roseflix?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix is a streaming service that offers a wide variety of award-winning TV shows, movies, anime, documentaries and more on thousands of internet-connected devices.

You can watch as much as you want, whenever you want, without a single commercial – all for one low monthly price. There's always something new to discover, and new TV shows and movies are added every week!
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        How much does Roseflix cost?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Watch Roseflix on your smartphone, tablet, Smart TV, laptop or streaming device, all for one fixed monthly fee. Plans range from ₱149 to ₱549 a month. No extra costs, no contracts.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        Where can I watch?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Watch anywhere, anytime, on an unlimited number of devices. Sign in with your Roseflix account to watch instantly on the web at roseflix.com from your personal computer or on any internet-connected device that offers the Roseflix app, including smart TVs, smartphones, tablets, streaming media players and game consoles.

You can also download your favourite shows with the iOS, Android, or Windows 10 app. Use downloads to watch while you're on the go and without an internet connection. Take Roseflix with you anywhere.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        How do I cancel?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix is flexible. There are no pesky contracts and no commitments. You can easily cancel your account online in two clicks. There are no cancellation fees – start or stop your account anytime.
      </div>
    </div>
    <div
      class="accordion__Item-sc-11jpbmb-3 bTuqNX"
    >
      <div
        class="accordion__Header-sc-11jpbmb-4 cLROnR"
      >
        What can I watch on Roseflix?
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </div>
      <div
        class="accordion__Body-sc-11jpbmb-5 cnhtEn"
      >
        Roseflix has an extensive library of feature films, documentaries, TV shows, anime, award-winning Roseflix originals, and more. Watch as much as you want, any time you want.
      </div>
    </div>
  </div>
</div>
`;
